src/components/buttonMenu/buttonMenu.svelte

```diff
@@ -31,6 +31,11 @@
     if (isOpen === undefined) isOpenInternal = toggleTo
     onChange?.({ isOpen: toggleTo })
   }
+
+  const handleClose = (e: CloseEventDetail) => {
+    if (isOpen === undefined) isOpenInternal = false
+    onClose?.(e)
```

Comment:
Should this also fire the onChange event?

Reply:
Nice catch! That's probably why the controlled version wasn't working!